dova/www/js/controllers.js

```diff
@@ -17,7 +17,55 @@ angular.module('app.controllers', [])
                 $rootScope.register_success = true;
                 $state.go('login');
             }, function errorCallback(response) {
+                var changed = false;//check if the error text has changed
+
+                if($rootScope.checkConnection()) changed = true;
+                
+                //password checks bools
+                var hasNumber = false;
+                //get tags
+                var errorText = document.getElementById("error-text");
+                var firstName = document.getElementById("firstNameField");
+                var lastName = document.getElementById("lastNameField");
+                var username = document.getElementById("usernameField");
+                var password = document.getElementById("passwordField");
+
+                errorText.innerHTML = "";//reset error text
+
+                if (firstName.value.length == 0){ 
+                    errorText.innerHTML += "First Name should not be left Empty! <br>";
+                    changed = true;
+                }
+
+                if (lastName.value.length == 0){ 
+                    errorText.innerHTML += "Last Name should not be left Empty! <br>";
+                    changed = true;
+                }
+
+                if (username.value.length < 3){ 
+                    errorText.innerHTML += "Username should be at least 3 characters long! <br>";
+                    changed = true;
+                }
+
+                if (password.value.length < 8){
+                    errorText.innerHTML += "Password should be at least 8 characters long! <br>";
+                    changed = true;
+                }
+
+                //check if a number is present in the password
+                var matches = password.value.match(/\d+/g);
+                if (matches != null){
+                    hasNumber = true;
+                    changed = false;
```

Comment:
If password has a digit in it only Signup Error will be displayed. I'm removing this part of logic because we don't need to concern about it too much. @marouane97